File: Stack/code.js
class Stack {
  constructor() {
    this.stk = []
  }

  isEmpty() {
    return (this.stk.length === 0)
  }

  length() {
    return this.stk.length
  }

  peek() {
    if (this.stk.length === 0) {
      throw new Error("Stack is Empty!")
    } else {
      return this.stk[(this.stk.length - 1)]
    }
  }

  pop() {
    if (this.stk.length === 0) {
      throw new Error("Stack is Empty!")
    } else {
      return this.stk.pop()
    }
  }

  push(value) {
    return this.stk.push(value)
  }

}

const myStack = new Stack()
console.log("myStack:", myStack);
console.log("isEmpty:", myStack.isEmpty())
myStack.push(10);
myStack.push(20);
myStack.push(30);
console.log("check Stack:", myStack.peek())
myStack.push(40);
console.log("pop element:",myStack.pop())
console.log("Stack Length:",myStack.length())
console.log("updated Stack:", myStack)

File: Stack/code.py
class stack:
    def __init__(self):
        self.s = []

    def length(self):
        return len(self.s)
    
    def push(self,value):
        self.s.insert(0,value)
    
    def pushAppend(self,value):
        self.s.append(value)

    def peek(self):
        if(len(self.s) == 0):
            raise Exception("Stack is Empty!")
        else:
            return self.s[0]
        
    def peek2(self):
        if(len(self.s) == 0):
            raise Exception("Stack is Empty!")
        else:
            return self.s[len(self.s)-1]
        
    def pop(self):
        if(len(self.s) == 0):
            raise Exception("Stack is Empty!")
        else:
            return self.s.pop(0)
        
    def popUsingPop(self):
        if(len(self.s) == 0):
            raise Exception("Stack is Empty!")
        else:
            return self.s.pop()
        
    
# stk = stack()
# stk.push(10)
# stk.push(20)
# stk.push(30)
# # print(stk.peek())
# print(stk.pop())
# print(stk.pop())
# print(stk.pop())


stk2 = stack()
stk2.pushAppend(10)
stk2.pushAppend(20)
stk2.pushAppend(30)
print(stk2.peek2())
print(stk2.popUsingPop())
print(stk2.popUsingPop())
print(stk2.popUsingPop())